strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/AuthenticatorsConfigController.java

```diff
@@ -28,63 +25,72 @@
 
-    @Inject
-    private AuthenticatorsScanner authenticatorsScanner;
+    private final AuthenticatorsRegistry authenticatorsRegistry;
+
+    private final AuthenticatorsScanner authenticatorsScanner;
+
+    public AuthenticatorsConfigController(AuthenticatorsRegistry authenticatorsRegistry,
+                                          AuthenticatorsScanner authenticatorsScanner)
+    {
+        this.authenticatorsRegistry = authenticatorsRegistry;
+        this.authenticatorsScanner = authenticatorsScanner;
+    }
 
     @ApiOperation(value = "Enumerates ordered collection of authenticators with order number and name")
-    @ApiResponses(value = { @ApiResponse(code = 200,
-            message = "The list was returned successfully.") })
-    @RequestMapping(value = "/", method = RequestMethod.GET, produces = { MediaType.APPLICATION_XML_VALUE,
-                                                                          MediaType.APPLICATION_JSON_VALUE })
-    public ResponseEntity<AuthenticatorsRegistry> list()
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "The list was returned successfully.") })
+    @GetMapping(value = "/",
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity list()
     {
         return ResponseEntity.ok(authenticatorsRegistry);
     }
 
     @ApiOperation(value = "Reorders authenticators by their indexes")
-    @ApiResponses(value = { @ApiResponse(code = 200,
-            message = "Reorder succeeded"),
-                            @ApiResponse(code = 500,
-                                    message = "Reorder failed") })
-    @RequestMapping(value = "/reorder/{first}/{second}",
-            method = RequestMethod.PUT)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = SUCCESSFUL_REORDER),
+                            @ApiResponse(code = 400, message = FAILED_REORDER) })
+    @PutMapping(value = "/reorder/{first}/{second}",
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = { MediaType.TEXT_PLAIN_VALUE,
+                         MediaType.APPLICATION_JSON_VALUE })
     public ResponseEntity reorder(@PathVariable int first,
-                                  @PathVariable int second)
+                                  @PathVariable int second,
+                                  @RequestHeader(HttpHeaders.ACCEPT) String acceptHeader)
     {
         try
         {
             authenticatorsRegistry.reorder(first, second);
-            return ResponseEntity.ok("Reorder succeeded");
+            return getSuccessfulResponseEntity(SUCCESSFUL_REORDER, acceptHeader);
         }
         catch (Exception e)
         {
-            logger.error("Error during reorder processing.", e);
-            return toError("Error during reorder processing: " + e.getLocalizedMessage());
+            return getBadRequestResponseEntity(FAILED_REORDER, acceptHeader);
```

Comment:
Using `getBadRequestResponseEntity` in this particular case is maybe not a good idea because the `Exception` does not get logged in the console/logs. Could you please replace it with `getExceptionResponseEntity(HttpStatus.BAD_REQUEST, FAILED_REORDER, e, acceptHeader)`?